.github/workflows/example-project.yml

```diff
@@ -80,3 +80,29 @@ jobs:
         working-directory: example-project/usage-from-c
         run: |
           make
+
+      - name: Install into temporary location
+        if: startsWith(matrix.os, 'windows') && matrix.toolchain-suffix == '-gnu'
+        working-directory: example-project
+        run: |
+          cargo cinstall --verbose --release --prefix=/mingw64 --destdir=temp
```

Comment:
You may want `shell: msys2 {0}` here as well so `/mingw64` gets expanded properly, unless that's not an issue

Reply:
Thanks, but I tried this and it didn't work.

I *think* the problem is that the string `/mingw64` is supposed to be written into the DLL literally, not in its expanded form.

That's why I added a separate copying step (using the `msys2` shell) where the path is indeed expanded correctly.

But of course I may be totally wrong about this ... I've tried quite a lot of different things ...

Reply:
The main issue I have with the string `/mingw64` being a literal one is that it only makes sense in msys2's stuff.

In our hypothetical test example project, if we were planning to use cargo-c's produced final files outside of msys2 and with other programs, the `/mingw64` string wouldn't make much sense since it's msys2's that is converting the paths to a proper windows path, as without it, `/mingw64` would mean "An absolute path from the root of the current drive." which can change meaning if you are in `D:\>` or `C:\>`, that is assuming the end project is used by something that would read that raw line and interpret it using Window's fopen etc.
If it's using msys2's mingw-w64-x86_64-pkgconf, it will probably be fine (msys2 often uses `sed` to replace the `prefix=` line with `prefix=${MINGW_PREFIX}` is the `prefix=` line is often loaded with an absolute path from GitHub action's CI and using pkgconf will be able to expand that path properly in the end user's layout)

Reply:
I agree that it would look cleaner, but I simply couldn't make it work.

I just tried it again to make sure: https://github.com/mgeier/cargo-c/commit/d12fe0efdba4f76edf94a7f4a3db881325d8f745

Here's the result: https://github.com/mgeier/cargo-c/runs/2512240535

```
cc     run_tests.c  -lexample_project -o run_tests
run_tests.c:1:10: fatal error: example_project.h: No such file or directory
    1 | #include <example_project.h>
      |          ^~~~~~~~~~~~~~~~~~~
```

If you know a way to make this work, please let me know.

But maybe it doesn't matter anyway if the wrong string is embedded in the DLL, I'm not sure if Windows even checks this?

Reply:
try adding in a `command -v cc` and see what it prints out, if it prints out `/usr/bin/cc` or similar, you might need https://github.com/lu-zero/cargo-c/pull/191#discussion_r626675935

Reply:
It prints `/mingw64/bin/cc`, is this good or bad?

Reply:
Should be good, can you try `find /mingw64 -name example_project.h` next? Wasn't it mentioned somewhere needing to pass pkg-config stuff?

Reply:
> `find /mingw64 -name example_project.h`

This returns nothing.

> Wasn't it mentioned somewhere needing to pass pkg-config stuff?

I'm still trying to do all this without using pkg-config (and the current state of this PR runs successfully without needing pkg-config).

Reply:
> > `find /mingw64 -name example_project.h`
> 
> This returns nothing.

Feels like it's not installing to the right place, but oh well, not sure if I can help much more than that 

running
```bash
cargo build
cargo cinstall --prefix=/mingw64
```
and `find /mingw64/ -name example*.h`
returns `/mingw64/include/example_project.h`
so it seems it works on my machine, and running
```bash
make -C usage-from-c
```
compiles just fine
find

Reply:
> Feels like it's not installing to the right place

Is this something that can/should be fixed in `cargo-c`?

> In our hypothetical test example project, if we were planning to use cargo-c's produced final files outside of msys2 and with other programs, the `/mingw64` string wouldn't make much sense

Just out of curiosity I tried this in 7cb6e8e4578d608c07e9d601f54414754bbe7568 where I'm not setting `--prefix` (i.e. it is using the default `/usr/local`), and this seems to work fine nevertheless.

So it looks like this is indeed ignored on Windows (or not even written into the DLL?).